src/Toggle/Segment/SegmentPolicyEvaluator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Clearbooks\Labs\Client\Toggle\Segment;
+
+use Clearbooks\Labs\Client\Toggle\Gateway\SegmentTogglePolicyGateway;
+
+class SegmentPolicyEvaluator
+{
+    /**
+     * @var SegmentPriorityArranger
+     */
+    private $segmentPriorityArranger;
+
+    /**
+     * @var SegmentTogglePolicyGateway
+     */
+    private $segmentTogglePolicyGateway;
+
+    /**
+     * @param SegmentPriorityArranger $segmentPriorityArranger
+     * @param SegmentTogglePolicyGateway $segmentTogglePolicyGateway
+     */
+    public function __construct( SegmentPriorityArranger $segmentPriorityArranger,
+                                 SegmentTogglePolicyGateway $segmentTogglePolicyGateway )
+    {
+
+        $this->segmentPriorityArranger = $segmentPriorityArranger;
+        $this->segmentTogglePolicyGateway = $segmentTogglePolicyGateway;
+    }
+
+    /**
+     * @param array $segments
+     * @param string $toggleName
+     * @return bool|null
+     */
+    public function evaluateSegmentPoliciesForToggle( $toggleName, array $segments )
```

Comment:
So I am guessing it looks for segments if they are enabled or not and if you have no segments then returns null? 

Why is that a separate case? 

I don't really want you to rewrite it just asking :)